Gems/GradientSignal/Code/Source/UI/GradientPreviewWidget.cpp

```diff
@@ -8,15 +8,46 @@
 
 #include <UI/GradientPreviewWidget.h>
 
+#include <QIcon>
 #include <QPainter>
+#include <QToolButton>
+#include <QVBoxLayout>
 
 namespace GradientSignal
 {
-    GradientPreviewWidget::GradientPreviewWidget(QWidget* parent)
+    GradientPreviewWidget::GradientPreviewWidget(bool enablePopout, QWidget* parent)
         : QWidget(parent)
     {
         setMinimumSize(256, 256);
         setAttribute(Qt::WA_OpaquePaintEvent); // We're responsible for painting everything, don't bother erasing before paint
+
+        // For the preview with popout icon, configure an icon in the top-right
+        // corner of our preview that only appears when the user hovers over
+        // the preview
+        if (enablePopout)
+        {
+            QVBoxLayout* layout = new QVBoxLayout(this);
+            layout->setContentsMargins(QMargins(2, 2, 2, 2));
+            layout->setAlignment(Qt::AlignTop | Qt::AlignRight);
+
+            QIcon icon;
+            icon.addPixmap(QPixmap(":/Application/popout-overlay.svg"), QIcon::Normal);
+            icon.addPixmap(QPixmap(":/Application/popout-overlay-hover.svg"), QIcon::Active);
+
+            m_popoutButton = new QToolButton(this);
+            m_popoutButton->setIcon(icon);
+            m_popoutButton->setAutoRaise(true);
+            m_popoutButton->setIconSize(QSize(24, 24));
+            m_popoutButton->setCursor(Qt::PointingHandCursor);
+            m_popoutButton->hide();
+            layout->addWidget(m_popoutButton);
+            QObject::connect(m_popoutButton, &QToolButton::clicked, this, &GradientPreviewWidget::popoutClicked);
+        }
+        // For the popped out preview, make sure its window always stays on top
+        else
+        {
+            setWindowFlag(Qt::WindowStaysOnTopHint, true);
```

Comment:
I suggest we move this to `GradientPreviewDataWidget`'s `popoutClicked` handler - while it's most likely true that this either has a popout button or it is in fact a popout, it doesn't seem like an assumption we necessarily want to bake into this widget.

Reply:
I went back and forth on this, I think you are right I will move it, thanks!